test/unit/org/apache/cassandra/index/sai/cql/DataModel.java

```diff
@@ -0,0 +1,438 @@
+                    .add("'WY', 10000000000, false, '2015-06-17',  97093.14,  2.7,  '192.146.215.91',   586107,   57,   5,       'Wyoming', '00:15:50', '2015-06-17T00:00:00', e8a3c287-78cf-46b5-b554-42562e7dcfb3, 2576612e-d17d-11e8-a8d5-f2801f1b9fd1, 2")
+                    .build();
+
+    String STATIC_INT_COLUMN = "entered";
+
+    List<Pair<String, String>> STATIC_COLUMNS =
+            ImmutableList.<Pair<String, String>>builder().add(Pair.create(STATIC_INT_COLUMN, CQL3Type.Native.INT.toString() + " static"))
+                                                         .addAll(NORMAL_COLUMNS).build();
+
+    List<String> STATIC_COLUMN_DATA = ImmutableList.of("1819, " + NORMAL_COLUMN_DATA.get(0),
+                                                       "1819, " + NORMAL_COLUMN_DATA.get(1),
+                                                       "1850, " + NORMAL_COLUMN_DATA.get(2),
+                                                       "1850, " + NORMAL_COLUMN_DATA.get(3),
+                                                       "1910, " + NORMAL_COLUMN_DATA.get(4),
+                                                       "1910, " + NORMAL_COLUMN_DATA.get(5),
+                                                       "1792, " + NORMAL_COLUMN_DATA.get(6),
+                                                       "1792, " + NORMAL_COLUMN_DATA.get(7),
+                                                       "1788, " + NORMAL_COLUMN_DATA.get(8),
+                                                       "1788, " + NORMAL_COLUMN_DATA.get(9),
+                                                       "1817, " + NORMAL_COLUMN_DATA.get(10),
+                                                       "1817, " + NORMAL_COLUMN_DATA.get(11),
+                                                       "1896, " + NORMAL_COLUMN_DATA.get(12),
+                                                       "1896, " + NORMAL_COLUMN_DATA.get(13),
+                                                       "1845, " + NORMAL_COLUMN_DATA.get(14),
+                                                       "1845, " + NORMAL_COLUMN_DATA.get(15));
+
+    static AtomicInteger seq = new AtomicInteger();
+
+    DataModel withTableOptions(String tableOptions) throws Throwable;
+
+    String indexedTable();
+
+    String nonIndexedTable();
+
+    List<Pair<String, String>> keyColumns();
+
+    void createTables(Executor tester) throws Throwable;
+
+    void createIndexes(Executor tester) throws Throwable;
+
+    void insertRows(Executor tester) throws Throwable;
+
+    void insertRowsWithTTL(Executor tester) throws Throwable;
+
+    List<Object> executeIndexed(Executor tester, String query, int fetchSize, Object... values) throws Throwable;
+
+    List<Object> executeNonIndexed(Executor tester, String query, int fetchSize, Object... values) throws Throwable;
+
+    public class BaseDataModel implements DataModel
```

Comment:
This is the only implementation of `DataModel`. If we don't have short term plans to have another implementation we should remove the interface.